src/main/java/bio/terra/service/snapshot/flight/create/AddFlightIdToSnapshotRequestStep.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.service.snapshotbuilder.SnapshotRequestDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class AddFlightIdToSnapshotRequestStep implements Step {
+  private final SnapshotRequestDao snapshotRequestDao;
+  private final UUID snapshotRequestId;
+
+  public AddFlightIdToSnapshotRequestStep(
+      SnapshotRequestDao snapshotRequestDao, UUID snapshotRequestId) {
+    this.snapshotRequestDao = snapshotRequestDao;
+    this.snapshotRequestId = snapshotRequestId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    snapshotRequestDao.updateFlightId(snapshotRequestId, context.getFlightId());
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    // we don't want to remove the flightId if the flight fails
+    return StepResult.getStepResultSuccess();
```

Comment:
this is design choice. do we want to leave this for tracking purposes? The flight id is also returned by the flight so the user may already have this information